src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -727,7 +727,12 @@ private BoundExpression RebindSimpleBinaryOperatorAsConverted(BoundBinaryOperato
             BoundExpression? left = null;
             while (stack.TryPop(out current))
             {
-                Debug.Assert(current.Right is BoundUnconvertedInterpolatedString);
+                var right = current.Right switch
```

Comment:
>var right = current.Right switch [](http://example.com/codeflow?start=16&length=32)

The local isn't used. However, it looks like the intent is to  use it in place of ```current.Right``` in the next statement. Given that the tests are passing, I assume we are missing tests for "interesting" scenarios.

Reply:
It's actually not necessary at the moment, but more through what I would call a fluke of how the binder works than through intentional design. `BindSimpleBinaryOperator` will convert both the left and the right to the operand types of the operator (`string` in this case), which will end up back in this path on the right side, ensuring that no `BoundUnconvertedInterpolatedString`s are in the final tree. With the way that the code is currently implemented, I don't believe there is a way to actually see an observable difference between doing the work here and letting a recursive call eventually make its way back through this path, but as I said that feels more like unintended consequence than intentional behavior, so I will fix this bug. I'll also add a test that would break, if the behavior weren't already eventually consistent.